### ISSUE TITLE
Remove delay before hiding Premium advert

### ISSUE THREAD

**Issue by marxjohnson** [MediaBrowser/Emby#2469]
When attempting to play a song through the web interface, an advert is displayed inviting me to sign up for a Premium account.  This is fine, I have tried Premium and decided that I don't need any of the features. However, when attempting to close the advert and play my song, I am forced to wait for 15 seconds before the "Continue" button becomes active.  This is a terrible user experience.  Could this delay be removed?

**Comment by NP-Hardass** [comment]
I have to agree.  This takes the web interface from freeware to nagware.  The default interface for Emby is the web interface.  There are currently advertisements around the web interface for a premium experience if someone opts to spend money to get additional features, but this goes a step too far.  Here, we go from an advertisement to a block in functionality.  Accidentally clicking outside of the advertisement, causes the 15 second timer to reset back to the full value.  The end result is that the user ends up being forced toward your payment option to be able to use the web interface freely.  If your intent is to make the default interface a pay to use interface, then you should just do that, rather than making it seem like a fully free interface, when it really isn't. You do risk losing almost all of your non-paying userbase by doing so.  If not, you should keep your commitment to keeping the web interface free.  This doesn't mean I'm condoning not having advertisements, but they should not impact the user who opts not to take advantage of them.  Once that happens, to a lesser degree, you get the effect of making the default interface premium only, which I am sure was not your intention. 

As an aside from the merits (or lack there of) of changing the relationship with free users, I personally will consider switching to another media server if this trend continues.  I use Emby because it is Free and Open Source Software, but consider nagware to be non-free in nature, and thus would start to consider other options.  In fact, I was considering adding official packaging for Gentoo Linux, but won't be doing so if Emby takes this stance moving forward.

**Comment by JohnArneBirkeland** [comment]
I must say that I also agree. The inclusion of the timer in the already free part, is dangerously close to moving Emby into the nagware category. As a result I am now seriously considering if I should fork and maintain a nag-free free version. I had a quick look, and the nag screen would be trivial to remove.

**Comment by joshuaboniface** [comment]
I have to agree as well. I've followed Emby with interest for over two years, and even through some problems (terrible browser transcoding performance regardless of quality and lack of LDAP [#1146] going on a year-and-a-half), I've wanted this project to succeed, even if I couldn't directly use it yet. Not a month ago after purchasing a Chromecase I finally decided to just finally set it up and get it working, so at least I could use it to cast. And it worked wonderfully. I like your software, both because it's gratis, and because it's libre, and because most alternatives are neither.

But now we're hit with nags.

I respect the creator's desire to make money off of his project. I have no problem buying premium licenses, if they're either cheap, or provide useful features. And I have no problem buying Apps to support your mobile development efforts. But to add in nags in the web interface, the core of your functionality, to extort us to purchase an expensive license that for most of us provides no compelling benefit, is an insult to your casual users.

I would have gladly spent money, even $99, to help your project if I had gotten it working as I have before the nagging. I'm an experienced sysadmin and have been looking for years for a piece of software to do what Emby does, within my limitations (and LDAP was and is still one, even with a Chromecast, so I can properly interface with my central user management). It is good software. But extortion is not the way to get paid, and this will drive users away - to Plex, or for me, back to NFS and VLC, and for some, a fork.

No one wants that. And I hate that so often this sort of misguided monetization attempt goes on (Subsonic for the latest example), and FLOSS software in general, not to mention many people, are hurt because of it. You do have strict FLOSS supporters who want to support you, so let us do so without making us feel like you have to be slimy about it. Please reconsider this decision and remove the nags.

Respectfully yours, Joshua.

**Comment by x041** [comment]
this destroys the functionality of the software as it is currently implemented. Put in an "advertisement" fine, but forcing a timeout value and resting it? I get that might be a bug, but this is terrible design which will serve only to push all but the most rabid fan users away. 

Gentle reminders are fine, but this as it is, is NOT a gentle reminder, it is terrible nag-ware, and IMO nag-ware is extortion-ware. 👎

**Comment by RaymondSchnyder** [comment]
Please remove this dialog screen everytime i want to play something. This is just awfull and bad for the user experience!

**Comment by vimalkvn** [comment]
Yes, being greeted by this and advertisements like the "Discover Emby Theater" in the main interface significantly degrades user experience of an otherwise brilliant product. Please consider removing them.

One possibility would be to highlight the Emby Premiere section under settings probably by
* Moving it to a higher level (same as Server, Extras etc., )
* Highlight premiere features using screenshots or videos under the Emby Premiere section.

**Comment by tkz79** [comment]
I not only also agree, I would also vote for people who pushed for adding this to be removed from any future decision making... Annoying people is not the way to get them to want to give you money, its a way to alienate them... 

I'd be willing to support you guys, but not for outrageous prices for "bonuses" i don't want nor need. 
$5 / mo $50 / year / 99 lifetime is ridiculous for a small app that includes NO content that i host myself. $5 / yr, $20 / life seems more realistic.

**Comment by untotren** [comment]
$100 for LIFETIME access really isn't that bad guys, cheaper than 2 xbox games and it helps feed the devs, cmon.

**Comment by marxjohnson** [comment]
I didn't raise this bug as an objection to the existence or cost of premium membership.  I raised it as an objection to the deliberate introduction of an anti-feature which degrades the user experience.

**Comment by Frozen-byte** [comment]
I agree that the reminder for buying Premium is not the issue.
It is the **15 seconds delay** which becomes terrible long in users experience.

**Comment by joshuaboniface** [comment]
I'll leave an update to say that I agree completely with @marxjohnson - this is not about the existence of the premium membership (even though I personally feel it's too high). This is about extortionist nagware which turns Emby from a free application with paid premium features, into nagware/trialware. As I said in my original post, such behaviour is a sure way to insult new users ("we don't care about your user experience unless you pay us NOW" - this is not a mentality conducive to collaborative software where the users are important too) and drive them away from a fantastic project.

I've recently updated to a later release (3.2.20.0) and I can confirm that Emby is no longer nagging before *ever* video as it was before. In fact, I haven't seen one yet, though I've been using it less than planned. So I'd say someone listened even if they're not posting it here.

**Comment by marxjohnson** [comment]
I've just updated to 3.2.20.0 and I am still seeing the delay when trying to play music.

**Comment by RaymondSchnyder** [comment]
Its possible to drive a fork without these nagging ads, if the devs arent gonna respond to this! Mark them or they will continue to ignore this problem

**Comment by nvllsvm** [comment]
No response on fixing this, bah!

@RaymondSchnyder - I've essentially forked and made both an Arch Linux package and a Docker image of Emby patched to remove the nagware.

**Arch Linux**: https://aur.archlinux.org/packages/emby-server-unlocked/
**Docker**: https://hub.docker.com/r/nvllsvm/emby-unlocked/

**Comment by nvllsvm** [comment]
Seems my fork wasn't welcome on their forum as my response to a related thread was removed.

https://emby.media/community/index.php?/topic/44273-being-nagged-to-purchase-emby-premiere/page-10

**Comment by schoerg** [comment]
Well of course they don't like it. It potentially drives users away from paying. Although, on the other side it might drive users away from Emby because of this.

**Comment by nvllsvm** [comment]
@schoerg - On the flip side, increasingly turning Emby into nagware is driving people like me closer towards a legitmate fork.

**Comment by schoerg** [comment]
There was a fork like this some years ago, for the same issue. The problem is that with time, the fork becomes less and less maintained and will lack behind.
I am not saying this will happen with your fork (not enough time has passed), however it has happened with the other one I know of.

**Comment by nvllsvm** [comment]
My current fork is easy to maintain. It's the exact same PKGBUILD as emby-server in the Arch repos, but with an added patch:
https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=emby-server-unlocked#n36

The Docker image is forked from the binhex/emby image, but builds and installs my version instead. Also easy to maintain.

**Comment by joshuaboniface** [comment]
@nvllsvm Thanks for that, I've used your patch to build a de-naggified Debian package as well - extremely straightforward, just had to convert it to a unified diff and add it to the `debian/patches` directory in the `emby-server` source package. I don't think this issue is going to ever be "resolved" officially, especially considering the opinions expressed in the echo-chamber forum thread about this, so we're on our own here.

**Comment by ebr11** [comment]
> ("Lifetime" subscription is not a lifetime subscription)

From our License:

Lifetime Term: Lifetime shall mean the entire period in which the respective product, services, and/or features, as indicated below and updated from time to time, are offered for license.

**Comment by schoerg** [comment]
Thanks for this. I remember that it was something like this:

> Lifetime Term: Lifetime shall mean the current major version of the Emby server software or twenty-four months, whichever is longer.  (E.g.  The Emby server software is currently at version 3.XXXX. A lifetime subscription shall be valid until Emby server software reaches version 4.XXX or twenty-four months, whichever occurs later.)

**Comment by LukePulverenti** [comment]
That was quite a while ago, and we adjusted based on community feedback. Please check out our newer terms. Thanks.

**Comment by nvllsvm** [comment]
Solution to this is available here: https://github.com/nvllsvm/emby-unlocked

**Comment by nvllsvm** [comment]
@untotren - Price has been raised to $119 for lifetime.

**Comment by nvllsvm** [comment]
@ebr11 , @LukePulverenti - If I supply a PR to remove the nag screen, are you willing to merge?

**Comment by nvllsvm** [comment]
https://github.com/MediaBrowser/emby-webcomponents/pull/4 has been opened to remove the delay from the periodic message.

**Comment by nvllsvm** [comment]
I'm beginning to think that @ebr11 and @LukePulverenti have **zero** intentions of ever removing the nagscreen. Is this issue only open to serve as an echo chamber for the sake of placating those annoyed by the nag screen?

Give us something please. This issue has been open since early 2017 without any hint of resolution. If you never want to remove it, alright. We understand, you work hard on Emby and it supports your lives. 

Be honest - either acknowledge you're willing to remove it or close this issue. Stop toying with us.

**Comment by deten** [comment]
@nvllsvm Do you know if that patch will work on Synology NAS?

**Comment by nvllsvm** [comment]
@deten - Not sure. Try it out and file an issue at https://github.com/nvllsvm/emby-unlocked if you run into any problems.

**Comment by deten** [comment]
@nvllsvm Okay I am not familiar with building the package myself.  So I will give it a try but I don't know step one...

**Comment by nvllsvm** [comment]
@deten - I've only built an Arch Linux package and Docker image. Not familiar enough with the other package types to build them. If someone figures it out, I'm willing to add that to the emby-unlocked project.

**Comment by totallyfake** [comment]
I just got hit with the nagscreen for the first time and I am quite unhappy. I was even thinking of getting premium before this, but I certainly am not anymore. I was also going to create a slackbuild for emby since there isn't one yet, so I'll definitely include this unlock patch.

**Comment by stev-it** [comment]
I also dont like the nagscreen, but its quite fair to display it some times.
But this 15 seconds delay, and then the nagscreen shows EVERYTIME. That is TERRIBLE user experience and will gain you nothing but angry customers who WILL change to plex.

tldr: keep nagscreen but only show it a few times (not every time goddamit)  and remove the delay.

**Comment by jkorany** [comment]
Really wish the devs would respond to this

**Comment by schoerg** [comment]
They have. They want to make money with emby. You don't get to make money with GPL software, nobody is going to donate to keep development and/or site running.

**Comment by RaymondSchnyder** [comment]
But i dont see which they implement this stupid nagscreens if they want to make money, these are exactly those things who keeps future customers AWAY from emby.
Just do it like plex,
emby server & clientapplications opensource,
and a version of emby server & clientapplications which costs a bit, but it has more different futures or faster updates. (I dont mean restricting the opensource version, i mean just faster development on the "premium" one, and more features for premium users like LDAP, oder Two Factor...)

I know this solution is not gonna be appreciated, but its the best they could do.

**Comment by xInsertx** [comment]
I actually like that model from plex a lot better. It's just the company's direction that is pretty bad. I was planning on buying a lifetime membership but this seems pretty sketchy.

**Comment by nvllsvm** [comment]
Friendly update on a community soft-fork:

- emby-unlocked repo has 63 stars
- The emby-unlocked Docker image has been pulled over 8,700 times.

**Comment by charlienewey** [comment]
I would have genuinely been happy to purchase Emby Premium, because I feel that people should get paid for their work - particularly if it's such a well-written piece of software. However, I really don't feel I can support a development team that makes such user-unfriendly UX decisions - it shows a lack of awareness of one's user base and is - to be honest - outright hostile to users trying to experiment with Emby.

**Comment by lazy8** [comment]
Thanks for this thread, decision made. Bye.

**Comment by Skyost** [comment]
Hey @nvllsvm, can you please post a little manual on how to build it on Debian / Ubuntu ? I'm pretty new in the Linux world 😄

**Comment by nvllsvm** [comment]
@Skyost - Not sure how to build it. I recommend setting up Docker and using the emby-unlocked Docker image. Once Docker is setup, using the image should be even easier than compiling or using distro-specific packages.
https://hub.docker.com/r/nvllsvm/emby-unlocked/

**Comment by nvllsvm** [comment]
Friendly update on a community soft-fork:

- [emby-unlocked](https://github.com/nvllsvm/emby-unlocked) repo has 124 stars
- The [emby-unlocked Docker image](https://hub.docker.com/r/nvllsvm/emby-unlocked/) has been pulled over 100,000 times.

**Comment by LukePulverenti** [comment]
I haven't mentioned this but in recent months this advert is shown much less frequently. In exchange for the occasional advertisement you're getting user support that you're going to have a hard time finding elsewhere.

**Comment by marxjohnson** [comment]
@LukePulverenti Thanks for taking the time to respond.  If I had the choice I'd take seeing the advert every time with no delay rather than seeing it sometimes with a delay, but I appreciate your position differs.

I'm going to close the issue here as it seems the discussion has run its course.

**Comment by LukePulverenti** [comment]
New release is beginning to roll out. No more pre-playback advertisement. Thanks guys.

**Comment by TheSIlentLink** [comment]
@LukePulverenti this release has no macOS version or has it not been built yet?

**Comment by marxjohnson** [comment]
Thanks very much, I really appreciate you making this change, as I'm sure do others in this thread.

**Comment by LukePulverenti** [comment]
It's gradually rolling out for each platform. Your server dashboard will tell you when it's available, thanks.

**Comment by TheSIlentLink** [comment]
Thank you very much for listening to the community feedback and making the change.
